lib/resources/host.rb

```diff
@@ -121,12 +121,20 @@ def resolve(hostname)
   end
 
   class LinuxHostProvider < HostProvider
-    # ping is difficult to achieve, since we are not sure
-    def ping(hostname, _port = nil, _proto = nil)
-      # fall back to ping, but we can only test ICMP packages with ping
-      # therefore we have to skip the test, if we do not have everything on the node to run the test
-      ping = inspec.command("ping -w 1 -c 1 #{hostname}")
-      ping.exit_status.to_i != 0 ? false : true
+    def ping(hostname, port = nil, proto = nil)
+      # assume tcp if port is given without corresponding protocol
+      if port && !proto
+        proto = 'tcp'
+      end
+      # use bash builtin capability for checking connectivity
+      # for those os's with timeout from coreutils packages
+      # TODO: Verify whether this works on other linux versions
+      if port && (inspec.os.redhat? || inspec.os.debian?)
+        resp = inspec.command("timeout 1 bash -c 'cat </dev/null >/dev/#{proto}/#{hostname}/#{port}'")
```

Comment:
That would probably be safe, but I thought for some reason it was nearly universal in inclusion. At least on redhat and debian families. Is that not the case. By the same token we are going to be safe with bash, we should probably do the same thing for the timeout command, which is in both redhat coreutils and debian coreutils packages. And were you suggesting we'd modify as:
~~~
if port && (inspect.os.redhat? || inspec.os.debian?) && 
   command('bash').exist? && command('timeout').exist?
~~~
Or were you thinking of a different modification.